oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -1195,13 +1191,16 @@ impl Runtime {
         // below, because that takes ownership of the instance itself.
         //
         // We also want no locks to be held while the instance is starting.
-        let node_join_handle = match node_instance.start() {
-            Ok(join_handle) => join_handle,
-            Err(status) => {
-                self.remove_node_id(node_id);
-                return Err(status);
-            }
-        };
+        let runtime = self.clone();
+        let node_join_handle = thread::Builder::new()
+            .name(node_name.to_string())
+            .spawn(move || {
+                node_instance.run();
+                // It's now safe to remove the state for this Node, as there's nothing left
+                // that can invoke `Runtime` functionality for it.
+                runtime.remove_node_id(node_id)
```

Comment:
It seems a bit odd to have this removal logic here, I guess it's fine, but I wonder if it should be somewhere else (e.g. just in the runtime termination logic). Did you consider alternatives?

Reply:
TBH, having the node removal logic anywhere *other* than here seems odd to me.  This is the point where the Node's thread completes, so having it anywhere would require gratuitous inter-thread synchronization, and deferring it to runtime termination would mean that `node_infos` has unbounded growth.